Core/Core/SubmitAssignmentExtension/Model/FilePreviewProvider.swift

```diff
@@ -36,10 +36,9 @@ public class FilePreviewProvider {
 
     /** Starts the asynchronous loading of preview data from the given url. */
     public func load() {
-        DispatchQueue.global().async { [weak self] in
+        Task.detached { [weak self] in
```

Comment:
Maybe this feature could be highlighted in the test plan in the PR description to make sure it's QA-ed